app/models/voucher.rb

```diff
@@ -18,6 +20,9 @@ class Voucher < ApplicationRecord
 
   TYPES = ["Vouchers::FlatRate", "Vouchers::PercentageRate"].freeze
 
+  scope :vine, -> { where(voucher_type: VINE_TYPE) }
+  scope :not_vine, -> { where.not(voucher_type: VINE_TYPE).or(where(voucher_type: nil)) }
```

Comment:
If voucher_type is null, then it is definitely not "VINE", so I don't think the extra `or` is needed here, is it?

Reply:
I did not think it was needed, but testing told me otherwise. I revisited and figured out why :
https://www.postgresql.org/docs/current/functions-comparison.html


>  Ordinary comparison operators yield null (signifying “unknown”), not true or false, when either input is null. For example, 7 = NULL yields null, as does 7 <> NULL. When this behavior is not suitable, use the IS [ NOT ] DISTINCT FROM predicates:

I rewrote the scope with IS DISTINCT FROM

Reply:
Oh man, that's confusing! Thanks for testing.